doc/src/guides/solving/solve-ode.md

```diff
@@ -0,0 +1,423 @@
+using {func}`~.dsolve`:
+
+```py
+>>> from sympy import Function, dsolve, Derivative
+>>> from sympy.abc import x
+>>> y = Function('y')
+>>> result = dsolve(Derivative(y(x), x, x) + 9*y(x), y(x))
+>>> result
+Eq(y(x), C1*sin(3*x) + C2*cos(3*x))
+```
+
+You can then use SymPy to verify that the solution is correct:
+
+```py
+>>> from sympy import checkodesol
+>>> solution = result.rhs
+>>> solution
+C1*sin(3*x) + C2*cos(3*x)
+>>> checkodesol(Derivative(y(x), x, x) + 9*y(x), solution)
+(True, 0)
+```
+
+The output of {func}`~.checkodesol` is a tuple where the first item, a boolean,
+tells whether substituting the solution into the ODE results in `0`, indicating
+the solution is correct.
+
+## Guidance
+
+### Input Format
+
+There are many ways to express derivatives of functions. For an undefined
+function, both {class}`~.Derivative` and {func}`~.diff` represent the undefined
+derivative. Thus, all of the following `ypp` ("y prime prime") represent $y''$,
+the second derivative with respect to $x$ of a function $y(x)$:
+
+```py
+ypp = y(x).diff(x, x)
+ypp = y(x).diff(x, 2)
+ypp = y(x).diff((x, 2))
+ypp = diff(y(x), x, x)
+ypp = diff(y(x), x, 2)
+ypp = Derivative(y(x), x, x)
+ypp = Derivative(y(x), x, 2)
+ypp = Derivative(Derivative(y(y(x), x), x)
+ypp = diff(diff(y(x), x), x)
+yp = y(x).diff(x)
+ypp = yp.diff(x)
+```
+
+We recommend specifying the function to be solved for, as the second argument to
+{func}`~.dsolve`. Note that it must be a function rather than a variable
+(symbol). SymPy will give an error if you specify a variable ($x$) rather than a
+function ($f(x)$):
+
+```py
+>>> dsolve(Derivative(y(x), x, x) + 9*y(x), x)
+Traceback (most recent call last):
+    ...
+ValueError: dsolve() and classify_ode() only work with functions of one variable, not x
+```
+
+Similarly, you must specify the argument of the function: $f(x)$, not just $f$.
+
+You can define the function to be solved for in two ways. The subsequent syntax
+for specifying initial conditions depends on your choice.
+
+## Define a Function Without Including Its Independent Variable
+
+As in the example above, you can define a function without including its
+independent variable:
+
+```py
+>>> from sympy import symbols, Eq, Function, dsolve
+>>> f, g = symbols("f g", cls=Function)
+>>> x = symbols("x")
+>>> eqs = [Eq(f(x).diff(x), g(x)), Eq(g(x).diff(x), f(x))]
+>>> dsolve(eqs, [f(x), g(x)])
+[Eq(f(x), -C1*exp(-x) + C2*exp(x)), Eq(g(x), C1*exp(-x) + C2*exp(x))]
+```
+
+Note that you supply the functions to be solved for as a list as the second
+argument of {func}`~.dsolve`, here `[f(x), g(x)]`.
+
+### Specify Initial (Boundary) Conditions
+
+If your differential equation(s) have initial or boundary conditions, specify
+them with the {func}`~.dsolve` optional argument `ics`. It should be given in
+the form of `{f(x0): y0, f(x).diff(x).subs(x, x1): y1}` and so on where, for
+example, the value of `f(x)` at $x = $ `x0` is `y0`. For power series solutions,
```

Comment:
Don't mix LaTeX and code like this. Use either one or the other.

Reply:
Should also note here that `f'(x1) = y1` (that's the one that is less clear).

Reply:
My interpretation was that

- `code` is for code, e.g. `dsolve`, as in the [documentation style guide](https://docs.sympy.org/dev/contributing/documentation-style-guide.html#code)
- LaTeX is for math, e.g. $\pi$, similar to the [docstring style guide](https://docs.sympy.org/dev/contributing/docstring.html#docstrings-for-classes-that-are-mathematical-functions)

Does that make sense?

Reply:
For e.g. x = x0, make 0 a subscript. Use math.